runtime/logging/logger.go

```diff
@@ -38,100 +38,98 @@ type Options struct {
 	Weavelet   string // Service Weaver weavelet id (e.g., "36105c89-85b1...")
 
 	// Pre-assigned attributes. These will be attached to each log entry
-	// generated by the logger. This slice will never be appended to
-	// in place.
+	// generated by the logger. This slice will never be appended to in place.
 	Attrs []string
 }
 
-// makeEntry returns an entry that is fully populated with the provided level,
-// msg, and attributes. The file and line information is taken from the call
-// stack after skipping frameskip entries up. For example, a frameskip of 0
-// will use the file and line of the call to makeEntry, and a frameskip of 1
-// will use the file and line of the caller's parent.
-func makeEntry(level, msg string, attrs []any, frameskip int, opts Options) *protos.LogEntry {
+// LogHandler implements a custom slog.Handler.
+type LogHandler struct {
+	slog.Handler
+	Opts  Options                      // configures the log entries
+	Write func(entry *protos.LogEntry) // called on every log entry
+}
+
+// Handle implements the slog.Handler interface.
+func (h *LogHandler) Handle(rec slog.Record) error {
+	h.Write(h.makeEntry(rec))
+	return nil
+}
+
+// Enabled implements the slog.Handler interface.
+func (h *LogHandler) Enabled(context.Context, slog.Level) bool {
+	// Support all logging levels.
+	return true
+}
+
+// WithAttrs implements the slog.Handler interface.
+func (h *LogHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
```

Comment:
Is `WithAttrs` supposed to return a copy of the handler and leave the original handler unmodified? For example, consider the following code: 

```go
logger := ...
logger.WithAttrs([]slog.Attry{{"foo", slog.StringValue("bar")}})
logger.Debug("hi")
```

With this code, it looks like the `Debug` log would have the `"foo"="bar"` attribute, but I would expect it not to? I could be misunderstanding the `Handler` interface though.